### PR TITLE
Fix SPV IR for generic opaque types

### PR DESCRIPTION
**Pull request by Fznamznon**
This patch fixes translation of OpTypeQueue, OpTypeEvent,
OpTypeDeviceEvent, OpTypeReserveId and OpTypeSampler so they are not
traslated directly to OpenCL in case SPIR-V friendly IR output is
requested.
This patch also fixes SPV IR for OpEnqueueMarker and OpGetDefaultQueue.

### CODE REVIEW

**Comment by Fznamznon** [comment]
@aratajew , @Nuullll , @AlexeySotkin , @MrSidims , I'd appreciate if you take a look

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -1511,9 +1507,10 @@ void SPIRVToLLVM::transGeneratorMD() {
 
 Value *SPIRVToLLVM::oclTransConstantSampler(SPIRV::SPIRVConstantSampler *BCS,
                                             BasicBlock *BB) {
-  auto *SamplerT =
-      getOrCreateOpaquePtrType(M, OCLOpaqueTypeOpCodeMap::rmap(OpTypeSampler),
-                               getOCLOpaqueTypeAddrSpace(BCS->getOpCode()));
+  std::string Name = kSPIRVTypeName::PrefixAndDelim +
```
Comment (Nuullll) [inline lib/SPIRV/SPIRVReader.cpp:1510]: NIT: could we encapsulate this logic into some function like `getSPIRVOpaquePtrType` to reduce duplicated code a litte bit?

```cpp
PointerType *getSPIRVOpaquePtrType(Module *M, Op OC) {
  std::string Name = getSPIRVTypeName(SPIRVOpaqueTypeOpCodeMap::rmap(OC));
  return getOrCreateOpaquePtrType(M, Name, getOCLOpaqueTypeAddrSpace(OC));
}
```

**Review by MrSidims — APPROVED** [review]
LGTM

File: lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
Comment (Fznamznon) [reply lib/SPIRV/SPIRVReader.cpp]: Thank you, done.

**Review by Nuullll — APPROVED** [review]
LGTM

**Review by svenvh — APPROVED** [review]

**Pull request merged**